File: Week_10/Practicum/Find Center of Star Graph.py
class Solution:
    def findCenter(self, edges: List[List[int]]) -> int:
        in_degree = defaultdict(int)

        for edge in edges:
            in_degree[edge[0]] += 1
            in_degree[edge[1]] += 1

            if in_degree[edge[0]] > 1:
                return edge[0]
            if in_degree[edge[1]] > 1:
                return edge[1]

File: Week_10/Practicum/Find Center of Star Graph.cpp
class Solution {
public:
    int findCenter(vector<vector<int>>& edges) {
        unordered_map<int, bool> visited;

        for (auto& e : edges) {
            if (visited.count(e[0]) != 0)
                return e[0];
            if (visited.count(e[1]) != 0)
                return e[1];
            visited[e[0]] = true;
            visited[e[1]] = true;
        }

        return -1;
    }
};